screens/HomeAuth.js

```diff
@@ -8,167 +8,152 @@ import {
-        const userType = JSON.parse(hmDone).user_type;
-        if (userType === 'mentor') {
-          headerTitle = 'Mentees';
-        }
-
-        this.props.navigation.navigate('menteeListView', { headerTitle });
-      } else {
-        console.log('Getting FB token');
-        const hmDone = await this.getHeyMentorToken(this.state.fbToken, 'facebook');
-
-        const userType = JSON.parse(hmDone).user_type;
-        if (userType === 'mentor') {
-          headerTitle = 'Mentees';
-        }
-
-        this.props.navigation.navigate('menteeListView', { headerTitle });
-      }
-    }
+    this.attemptLogin();
   }
 
   onButtonPress = () => {
     this.setState({ loading: true });
     this.facebookLogin();
   };
 
-  onAuthComplete = () => {
-    // after user successfully logs in navigate to menteeListView page
-    /*    if (this.state.fbToken) {
-      this.props.navigation.state = this.state;
-
-      this.props.navigation.navigate('menteeListView', {
-        fbId: this.state.fbUserId
-      });
-    } */
-  };
-
   getHeyMentorToken = async (token, authType) => {
-    const API_URL = 'http://ppeheymentor-env.qhsppj9piv.us-east-2.elasticbeanstalk.com';
-
-    console.log('Making GetToken request');
     console.log(`${API_URL}/register/${authType}?access_token=${token}`);
 
     const response = await fetch(
       `${API_URL}/register/${authType}?access_token=${token}`,
       { method: 'post' }
     );
-    const responseJson = await response.json();
 
-    console.log('Printing responsejson from GetToken');
-    console.log(responseJson);
-    console.log('done printing');
+    console.log(`Response: ${response}`);
 
-    if (responseJson && !responseJson.error) {
-      console.log('Setting state');
-      this.setState({ hmToken: responseJson });
-      await AsyncStorage.setItem('hm_token', JSON.stringify(responseJson));
-      console.log('Done setting state');
-    } else {
+    try {
+      const responseJson = await response.json();
+      if (responseJson && !responseJson.error) {
+        console.log('Saving HM token to AsyncStorage');
+        await AsyncStorage.setItem('hm_token', JSON.stringify(responseJson));
+      }
+      console.log('Got response from API: ');
+      console.log(responseJson);
+      return true;
+    } catch (e) {
       console.log('Error authenticating with fed token');
+      console.log(e);
     }
+    return false;
   };
 
+  async getFacebookData(token) {
+    // API call to FB Graph API. Will add more code to fetch social media data
+    const response = await fetch(
+      `https://graph.facebook.com/me?access_token=${token}`
+    );
+    const responseJson = await response.json();
+    console.log(`Facebook ID: ${responseJson.id}`);
+    this.fbId = `${responseJson.id}`;
+
+    return responseJson;
+  }
+
   facebookLogin = async () => {
-    this.initFacebookLogin();
+    const tokenPromise = this.initFacebookLogin();
```

Comment:
From below this could potentially return null, should we add a check here to make sure the init was successful?

Reply:
Yeah good catch